site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/repository/SiteStore.java

```diff
@@ -39,21 +36,28 @@ public SiteStore(FhirRepository repository,
      */
     @Override
     public String saveEntity(Site entity) {
-
-        // TODO: Check if the Organization already exists.
+        String orgId = "";
 
         Organization org = converter.convert(entity);
-        String orgId = repository.saveOrganization(org);
+        orgId = repository.saveOrganization(org);
         org.setId(orgId);
-        logger.info(FhirRepo.SAVE_REQUEST.message(), orgId);
 
         // TODO: Add research study only when needed.
-        String researchStudyId = createResearchStudy(org);
-        logger.info(FhirRepo.SAVE_REQUEST.message(), researchStudyId);
+        createResearchStudy(org);
 
         return orgId;
     }
 
+    /**
+     * Find Organization By ID from the FHIR store
+     *
+     * @param organizationName of the organization to search.
+     * @return Organization searched by name.
+     */
+    public Organization findOrganizationByName(String organizationName) {
```

Comment:
This means that the caller (the service layer) is exposed to the FHIR implementation since you return a FHIR org. See my comment below in the service class too... 
Perhaps a better separation/design would be to have the store expose **Site**?
Also, I believe the method name here can be generalised, maybe something like findEntityByName?

Reply:
Addressed as discussed
We can generalize the name later